lib/undercover/plugin.rb

```diff
@@ -1,33 +1,53 @@
+# frozen_string_literal: true
+
 module Danger
-  # This is your plugin class. Any attributes or methods you expose here will
-  # be available from within your Dangerfile.
-  #
-  # To be published on the Danger plugins site, you will need to have
-  # the public interface documented. Danger uses [YARD](http://yardoc.org/)
-  # for generating documentation from your plugin source, and you can verify
-  # by running `danger plugins lint` or `bundle exec rake spec`.
+  # Report missing coverage report using undercover and danger-undercover
   #
-  # You should replace these comments with a public description of your library.
+  # You have to use [undercover](https://github.com/grodowski/undercover) to gather
+  # undercover report and send the report to this plugin so that danger-undercover
+  # can use it.
   #
-  # @example Ensure people are well warned about merging on Mondays
+  # @example Report missing coverage report
   #
-  #          my_plugin.warn_on_mondays
+  #          undercover.report('coverage/undercover.txt')
   #
-  # @see  rafayet-monon/danger-undercover
-  # @tags monday, weekends, time, rattata
+  # @see  nimblehq/danger-undercover
+  # @tags ruby, code-coverage, simplecov, undercover, danger, simplecov-lcov
   #
   class DangerUndercover < Plugin
+    # Checks the file validity and warns if no file found
+    # if valid file is found then if there's no changes,
+    # shows the report as message in Danger.
+    # If there's reports then it shows the report as warning in danger.
+    # @return  [void]
+    #
+    def report(undercover_path, sticky: true)
+      if valid_file? undercover_path
+        report = File.open(undercover_path).read
+        if report.match(/No coverage is missing in latest changes/)
+          message(report, sticky: sticky)
+        else
+          warn(report, sticky: sticky)
+        end
+      else
+        fail('undercover: Data not found')
```

Comment:
How about being a bit more descriptive here:

```suggestion
        fail('Undercover: coverage report cannot be found.')
```